pywbem_mock/_wbemconnection_mock.py

```diff
@@ -995,6 +1009,32 @@ def _display_objects(obj_type, object_repo, namespace, cmt_begin, cmt_end,
                     else:
                         _uprint(dest, obj.tomof())
 
+    def register_provider(self, namespace, classname, provider):
+        """
+        Register the provider object for namespace and class.
+        TODO: The registration data must be in the providers somewhere since
+        there is no link back to this class from the providers.
```

Comment:
I don't think so. The providers should be light-weight because there are many user-written providers. The providers do not need to know that or where they are registered. For example, if a particular instance provider is registered for two classes, it will be called for these two classes and sees which class a call is for at the time it is called. I don't think it needs to know when handling class A that it is also registered for class B.

Reply:
If you want to register a provider for multiple classes we can do either:

1. Chang class to list of classes.

2, The user files a registration for each class.

DISCUSSION

Reply:
The implemented provider registration allows for lists of classes and lists of namespaces. That is fine with me, and I think this discussion is resolved. Let me know if you still want to discuss this.

Reply:
Happy to close DONE

Reply:
I agree. Marked this one resolved